caluma/caluma_form/models.py

```diff
@@ -273,24 +277,9 @@ def copy(self, family=None):
 
         # copy answers
         for source_answer in self.answers.all():
-            new_answer = new_document.answers.create(
-                question=source_answer.question,
-                value=source_answer.value,
-                meta=dict(source_answer.meta),
-                date=source_answer.date,
-            )
-
-            if source_answer.question.type == Question.TYPE_FILE:
-                new_answer.file = source_answer.file.copy()
-                new_answer.save()
-
-            # TableAnswer: copy AnswerDocument too
-            for answer_doc in AnswerDocument.objects.filter(answer=source_answer):
-                new_doc = answer_doc.document.copy(family=family)
-
-                AnswerDocument.objects.create(
-                    answer=new_answer, document=new_doc, sort=answer_doc.sort
-                )
+            new_answer = source_answer.copy(document_family=family)
+            new_answer.document = new_document
```

Comment:
How about adding a `to_document` parameter to the `Answer.copy()` method? This way you'd avoid double-saving the answer

Reply:
Great idea! Done.